code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -97,7 +101,7 @@
 		collar.forceMove(loc)
 		collar = null
 	master_commander = null
-	GLOB.simple_animal_list -= src
+	GLOB.simple_animals[AIStatus] -= src
 	return ..()

```

Comment:
```
/mob/living/simple_animal/Destroy()
	GLOB.simple_animals[AIStatus] -= src
	if (SSnpcpool.state == SS_PAUSED && LAZYLEN(SSnpcpool.currentrun))
		SSnpcpool.currentrun -= src

	if(nest)
		nest.spawned_mobs -= src
		nest = null

	var/turf/T = get_turf(src)
	if (T && AIStatus == AI_Z_OFF)
		SSidlenpcpool.idle_mobs_by_zlevel[T.z] -= src

	return ..()
```

Is what TG has. Of course we still need to keep the collar and `master_commander` stuff too.

Reply:
still needs `GLOB.simple_animals[AIStatus] -= src`